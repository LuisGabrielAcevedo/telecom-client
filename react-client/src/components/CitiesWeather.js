import React, { useState, useEffect } from "react";
import Select from "./common/Select";
import Card from "./common/Card";
import "./cities-weather.scss";
import Loading from "./common/Loading";
import { getWeatherByCity } from "../helpers";

function CitiesWeather() {
  const [data, setData] = useState([]);
  const [selected, setSelected] = useState([]);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    const searchData = async () => {
      setLoading(true);
      const weatherResp = await getWeatherByCity(selected);
      setLoading(false);
      setData(weatherResp);
    };
    searchData();
  }, [selected]);

  const cards = data.map((item, i) => <Card key={i} {...item} />);

  return (
    <div>
      <Select setSelected={setSelected} selected={selected} />
      <div className="cities-weather__items">
        {!loading ? cards : <Loading repeat={8} />}
      </div>
    </div>
  );
}

export default CitiesWeather;
