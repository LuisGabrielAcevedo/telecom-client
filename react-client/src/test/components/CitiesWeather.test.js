import React from "react";
import "@testing-library/jest-dom";
import { shallow, mount } from "enzyme";
import CitiesWeather from "src/components/CitiesWeather";
import { act } from "react-dom/test-utils";

describe("CitiesWeather Component tests", () => {
  const wrapper = shallow(<CitiesWeather />);
  it("should init the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("", async () => {
    const wrapper = mount(<CitiesWeather />);
    await act(async () => wrapper);
    wrapper.update();
  });
});
