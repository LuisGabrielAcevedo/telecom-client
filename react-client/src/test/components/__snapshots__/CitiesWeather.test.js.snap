// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesWeather Component tests should init the component 1`] = `
<div>
  <Select
    selected={Array []}
    setSelected={[Function]}
  />
  <div
    className="cities-weather__items"
  />
</div>
`;
